### PR TITLE
Fixes #20544 - SCSS assets should use .scss extension

### PR DESCRIPTION
**Pull request by dLobatog**
Without using the proper extension, new versions of sass + sprockets
won't compile assets.
It will simply not find the assets because they don't have the right
extension. Also, we have to add the stylesheet assets to sprockets paths
- which was done automatically before.

In any case - https://github.com/theforeman/foreman/pull/4734 is a dependency for this PR, as otherwise asset precompilation does not even run.

### CODE REVIEW

**Comment by ehelms** [comment]
So  @dLobatog this is saying that CSS files HAVE to be converted to `.scss` extensions even when they are NOT SCSS files? That seems wrong, not that I am against hacky work arounds.

File: lib/katello/engine.rb [hunk 1]
```diff
@@ -69,6 +69,7 @@ def find_assets(args = {})
       javascripts = find_assets(:type => 'javascripts')
       images = find_assets(:type => 'images')
       vendor_images = find_assets(:type => 'images', :vendor => true)
+      vendor_stylesheets = find_assets(:type => 'stylesheets', :vendor => true)
```
Comment (ehelms) [inline lib/katello/engine.rb]: Does this result in them ending up in public/ ?

**Comment by ehelms** [comment]
Maybe its simpler in that https://github.com/Katello/katello/blob/master/app/assets/stylesheets/katello/katello.scss#L10 needs to be adjusted to have the `.css` file extension?

**Comment by ehelms** [comment]
In my local testing, and reading up on pipeline, this is all I had to change:

```
diff --git a/app/assets/stylesheets/katello/katello.scss b/app/assets/stylesheets/katello/katello.scss
index a1cd604..08f1148 100644
--- a/app/assets/stylesheets/katello/katello.scss
+++ b/app/assets/stylesheets/katello/katello.scss
@@ -1,3 +1,10 @@
+/**
+ * = require "katello/jquery.loadmask.css"
+ * = require "katello/jquery-ui-1.8.11.custom.css"
+ * = require "katello/jquery.treeTable.css"
+ * = require "katello/ui.spinner.css"
+ */
+
 @import "katello/overrides";
 @import "katello/katello_base";
 @import "katello/look";
@@ -7,11 +14,6 @@
 @import "katello/widgets/tabs";
 @import "katello/katello_sprites";
 
-@import "katello/jquery.loadmask";
-@import "katello/jquery-ui-1.8.11.custom";
-@import "katello/jquery.treeTable";
-@import "katello/ui.spinner";
-
 @import "katello/contents";
 @import "katello/generic";
 @import "katello/notifications";
diff --git a/vendor/assets/stylesheets/katello/jquery.treeTable.scss b/vendor/assets/stylesheets/katello/jquery.treeTable.css
similarity index 100%
rename from vendor/assets/stylesheets/katello/jquery.treeTable.scss
rename to vendor/assets/stylesheets/katello/jquery.treeTable.css
```

@dLobatog would you consider updating your PR with this? or should I open one?

File: lib/katello/engine.rb (hunk 1 above)
Comment (dLobatog) [inline lib/katello/engine.rb]: Yes, sprockets passes this to http://github.com/sass/sass/  to find them

**Comment by ehelms** [comment]
@dLobatog can you move those files back to the `css` versions?  As I understand it, with asset pipeline `require` is what will embed a file into another. So in this case, we have pure CSS files, we need embedded so using require is the proper way vs using sass `import` and modifying the files to be SCSS files when they really are not.

**Comment by ehelms** [comment]
[test]

**Comment by ehelms** [comment]
[test]

**Comment by ehelms** [comment]
[test]

**Comment by ehelms** [comment]
[test]

**Comment by ehelms** [comment]
[test]

**Comment by ehelms** [comment]
@dLobatog My apologies, seems I was a bit wrong in changes I had you make before :/ I think we needed a combination of your move to `.scss` files and the `requires` changes. You will notice there are 6 tests that fail currently. After some testing and research, the `css` files need to just have an added extension to them. So basically making this change:

```
#	renamed:    vendor/assets/stylesheets/katello/jquery-ui-1.8.11.custom.css -> vendor/assets/stylesheets/katello/jquery-ui-1.8.11.custom.css.scss
#	renamed:    vendor/assets/stylesheets/katello/jquery.loadmask.css -> vendor/assets/stylesheets/katello/jquery.loadmask.css.scss
#	renamed:    vendor/assets/stylesheets/katello/jquery.treeTable.css -> vendor/assets/stylesheets/katello/jquery.treeTable.css.scss
#	renamed:    vendor/assets/stylesheets/katello/ui.spinner.css -> vendor/assets/stylesheets/katello/ui.spinner.css.scss
```

Tests passed locally for me as well as assets compiled.

**Comment by ehelms** [comment]
Merged as the linked PR above, thanks again @dLobatog !

**Pull request closed without merging**